var/spack/repos/builtin/packages/protobuf/package.py

```diff
@@ -41,6 +41,8 @@ class Protobuf(AutotoolsPackage):
     depends_on('libtool',  type='build')
     depends_on('m4',       type='build')
 
+    conflicts('%gcc@:4.4.7') # Requires c++11
```

Comment:
I think `:4.6` would be better. We know that 4.6 doesn't accept `--std=c++11`, but 4.7 should.